RFC 165 (#3615)
taskcluster/taskcluster: ui/docs/manual/deploying/anonymous-role.mdx

---
title: Anonymous Role
---

# Anonymous Role

A single scope `assume:anonymous` will be granted for all calls (with or without credentials).
The `anonymous` role is aimed at private Taskcluster installations. With this, it will
be possible to prevent any useful access to the Taskcluster deployment without credentials

> allowing for greater administrator control over "public" endpoints.

Reviewer: Maybe add a `## Private Deployments` here, for symmetry, suggesting leaving the role empty.

> ## Public endpoints

Reviewer: ```suggestion
## Public Deployments
```

> All unauthenticated API calls are granted the scope `assume:anonymous`.

Reviewer: ```suggestion
All API calls are granted the scope `assume:anonymous`.
```

> In order to have all unauthenticated API calls public, grant the `anonymous` role the following scopes:

Reviewer: ```suggestion
In order to allow unauthenticated API, grant the relevant scopes to the `anonymous` role.
A "public" deployment of Taskcluster, allowing unauthenticated read-only access, would grant the following scopes:
```

  - `auth:current-scopes`
  - `auth:expand-scopes`
  - `auth:get-client:*`
  - `auth:get-role:*`
  - `auth:list-clients`
  - `auth:list-roles`
  - `github:get-badge:*`
  - `github:get-repository:*`
  - `github:latest-status:*`
  - `github:list-builds`
  - `hooks:get:*`
  - `hooks:list-hooks:*`
  - `hooks:list-last-fires:*`
  - `hooks:status:*`
  - `index:find-task:*`
  - `index:list-namespaces:*`

>   - `index:list-tasks:*`
>   - `index:list-tasks-post:*`,
>   - `purge-cache:all-purge-requests`

Reviewer: ```suggestion
  - `index:list-tasks:*`
  - `purge-cache:all-purge-requests`
```

---
title: Anonymous Role
---

# Anonymous Role

A single scope `assume:anonymous` will be granted for all calls (with or without credentials).
The `anonymous` role is aimed at private Taskcluster installations. With this, it will
be possible to prevent any useful access to the Taskcluster deployment without credentials
allowing for greater administrator control over "public" endpoints.

## Public Deployments

All API calls are granted the scope `assume:anonymous`.
In order to allow unauthenticated API, grant the relevant scopes to the `anonymous` role.
A "public" deployment of Taskcluster, allowing unauthenticated read-only access, would grant the following scopes:

>   - `auth:current-scopes`

Reviewer 2: Now that I think about it, this list will grow/shrink over time. How can we make sure we keep it up-to-date? Should we be generating this list somehow?

Reviewer 3: At the moment there isn't anything automated to keep it up to date but it's definitely something we should think about. I'll go ahead and create an issue for that assuming it's fine to do this in a follow up.

Reviewer 3: Filed https://github.com/taskcluster/taskcluster/issues/3737.
